src/trace/entry.zig

```diff
@@ -1,245 +1,87 @@
 const std = @import("std");
-const logfmt = @import("logfmt.zig");
-const Level = @import("level.zig").Level;
-const Channel = @import("../sync/channel.zig").Channel;
+const sig = @import("../sig.zig");
+const trace = @import("lib.zig");
+
+const Level = trace.level.Level;
+const ScopedLogger = trace.log.ScopedLogger;
 const AtomicBool = std.atomic.Value(bool);
 
-pub const Entry = union(enum) {
-    channel_print: ChannelPrintEntry,
-    direct_print: DirectPrintEntry,
-    noop,
+pub fn NewEntry(comptime scope: ?[]const u8) type {
+    return Entry(struct {}, scope);
+}
 
-    const Self = @This();
+pub fn Entry(comptime Fields: type, comptime scope: ?[]const u8) type {
+    return struct {
+        logger: ScopedLogger(scope),
```

Comment:
This is a nice improvement 👍. I considered having the Entry contain the actual logger instead of it being the one responsible for sending the message for logging but ran into some issue I could not get to compile.